docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,369 @@
+to prevent unexpected behavior, you should not create it in advance. 
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution, your Teleport instances need
+permissions to create, read from, and write to buckets.
+
+</Admonition>
+
+## TLS credential provisioning 
+
+High-availability Teleport deployments require an automated system to fetch
+certificates from Let's Encrypt and provision Teleport instances with them. 
+
+If you are running a single instance of the Teleport Auth Service and Proxy
+Service, you can configure this instance to fetch credentials for itself from
+Let's Encrypt using the [ACME ALPN-01
+challenge](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01), where
+Teleport demonstrates that it controls the ALPN server at the HTTPS address of
+your Teleport Proxy Service.
+
+For high-availability deployments, however, it is much more efficient to use a
+separate system to request TLS credentials from Let's Encrypt and provision
+Teleport instances with them. In this case, with multiple Teleport instances
+behind a load balancer, you will need to use the [ACME
+DNS-01](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge)
+challenge to demonstrate domain name ownership to Let's Encrypt. In this
+challenge, your TLS credential provisioning system creates a DNS TXT record with
+a value expected by Let's Encrypt.
+
+In the configuration we are demonstrating in this guide, each Teleport instance
+expects TLS credentials for HTTPS to be available at the file paths
+`/etc/teleport-tls/tls.key` (private key) and `/etc/teleport-tls/tls.crt`
+(certificate).
+
+## DNS service
+
+Set up a DNS zone where you can create records for Teleport, e.g., an Amazon
+Route 53 hosted zone or Google Cloud DNS zone.
+
+You must create a DNS A record that points to the address of your load balancer. 
+
+(!docs/pages/includes/dns-app-access.mdx!)
+
+<Admonition title="Required permissions">
+
+The TLS credential provisioner we mentioned earlier needs permissions to manage
+DNS records in order to satisfy Let's Encrypt's DNS-01 challenge. If you are
+using cloud-managed solutions, you should use your cloud provider's RBAC system
+(e.g., AWS IAM) to grant a role to the Proxy Service to manage DNS records. 
+
+</Admonition>
+
+## Teleport instances
+
+Run the Teleport Auth Service and Proxy Service as a scalable group of compute
+resources, for example, a Kubernetes `Deployment`  or AWS Auto Scaling group.
+This requires running the `teleport` binary on each Kubernetes pod or virtual
+machine or in your group. In the [Configuration](#configuration) section, we
+will show you how to configure each binary for high availability.
+
+### Open ports
+
+Ensure that, on each Teleport instance, the following ports allow traffic from
+the load balancer. If you do not plan to run a service associated with a given
+port, you can plan to leave that port closed.
```

Comment:
There are two modes to expose Teleport:
- separate listener which is the legacy way of doing, is super confusing, but most users are using it. This is the one described in this guide.
- multiplexing: the "new" one, one port to rule them all: 443 (or 3080). Ideally we would like to steer users toward the multiplexing mode (it's not the default for Helm charts because a lot of users are using a L7 lb for the 443 port and we don't support yet to pass through, hopefully this should happen in the next months and our life will become easier)

Do we want this section to reflect both modes? If we do, we might want to put multiplexing forward as it's easier for everyone.

Reply:
On the implementation team side, we generally recommend users opt for TLS routing unless there is an overriding reason to not use TLS routing as it simplifies the configuration of the infra and minimizes the port exposure of Teleport

Reply:
I personally don't really ever recommend TLS routing because >75% of the people I talk to want to put a load balancer of some kind in front of Teleport. >50% of the people want to use ACM. Lots of them want to use a reverse proxy like nginx or Traefik.

Probably around 10-15% want to use their own certs, and the rest are OK with Let's Encrypt. It's very popular, but just not a viable option in many environments.

This will definitely change if and when it becomes easier to keep ALPN/SNI intact behind load balancers.

Reply:
Since the question of whether to enable TLS Routing is implementation specific, I'll describe both approaches in this guide!